server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/DKIMSignTest.java

```diff
@@ -0,0 +1,343 @@
+        mci
+                .setProperty(
+                        "privateKey",
+                        TESTING_PEM);
+
+        mailet.init(mci);
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(mm)
+            .build();
+
+        Mailet m7bit = new ConvertTo7Bit();
+        m7bit.init(mci);
+
+        mailet.service(mail);
+
+        m7bit.service(mail);
+
+        ByteArrayOutputStream rawMessage = new ByteArrayOutputStream();
+        mail.getMessage().writeTo(rawMessage);
+
+        MockPublicKeyRecordRetriever mockPublicKeyRecordRetriever = new MockPublicKeyRecordRetriever(
+                "v=DKIM1; k=rsa; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDYDaYKXzwVYwqWbLhmuJ66aTAN8wmDR+rfHE8HfnkSOax0oIoTM5zquZrTLo30870YMfYzxwfB6j/Nz3QdwrUD/t0YMYJiUKyWJnCKfZXHJBJ+yfRHr7oW+UW3cVo9CG2bBfIxsInwYe175g9UjyntJpWueqdEIo1c2bhv9Mp66QIDAQAB;",
+                "selector", "example.com");
+
+        verify(rawMessage, mockPublicKeyRecordRetriever);
+    }
+
+    @Test
+    public void testDKIMSignMessageAsObjectConvertedTo7Bit()
+            throws MessagingException, IOException, FailException {
+        MimeMessage mm = new MimeMessage(Session
+                .getDefaultInstance(new Properties()));
+        mm.addFrom(new Address[]{new InternetAddress("io@bago.org")});
+        mm.addRecipient(RecipientType.TO, new InternetAddress("io@bago.org"));
+        mm.setContent("An 8bit encoded body with \u20ACuro symbol.",
+                "text/plain; charset=iso-8859-15");
+        mm.setHeader("Content-Transfer-Encoding", "8bit");
+        mm.saveChanges();
+
+        FakeMailContext FakeMailContext = FAKE_MAIL_CONTEXT;
+        FakeMailContext.getServerInfo();
+        FakeMailetConfig mci = new FakeMailetConfig("Test", FakeMailContext);
+        mci.setProperty(
+                "signatureTemplate",
+                "v=1; s=selector; d=example.com; h=from:to:received:received; a=rsa-sha256; bh=; b=;");
+        mci.setProperty(
+                "privateKey",
+                TESTING_PEM);
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(mm)
+            .build();
+
+        Mailet mailet = new DKIMSign();
+        mailet.init(mci);
+
+        Mailet m7bit = new ConvertTo7Bit();
+        m7bit.init(mci);
+        m7bit.service(mail);
+
+        mailet.service(mail);
+
+        m7bit.service(mail);
+
+        ByteArrayOutputStream rawMessage = new ByteArrayOutputStream();
+        mail.getMessage().writeTo(rawMessage);
+
+        MockPublicKeyRecordRetriever mockPublicKeyRecordRetriever = new MockPublicKeyRecordRetriever(
+                "v=DKIM1; k=rsa; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDYDaYKXzwVYwqWbLhmuJ66aTAN8wmDR+rfHE8HfnkSOax0oIoTM5zquZrTLo30870YMfYzxwfB6j/Nz3QdwrUD/t0YMYJiUKyWJnCKfZXHJBJ+yfRHr7oW+UW3cVo9CG2bBfIxsInwYe175g9UjyntJpWueqdEIo1c2bhv9Mp66QIDAQAB;",
+                "selector", "example.com");
+        verify(rawMessage, mockPublicKeyRecordRetriever);
+    }
+
+    @Test
```

Comment:
Moving is one issue, refactoring is an other. Do we want to take time doing this?

(It is an external contribution, we can ask Shrumit to do it...)